packages/eslint-plugin/src/rules/no-unsafe-unary-minus.ts

```diff
@@ -0,0 +1,55 @@
+import type * as ts from 'typescript';
+
+import * as util from '../util';
+
+interface TypeChecker extends ts.TypeChecker {
+  // https://github.com/microsoft/TypeScript/issues/9879
+  isTypeAssignableTo(source: ts.Type, target: ts.Type): boolean;
+  getUnionType(types: ts.Type[]): ts.Type;
```

Comment:
Oh 😞 this might be a problem for us. We try very hard not to use TypeScript-internal APIs in our rules. While these type assignability APIs have coincidentally been stable for quite some time, we've tried to be good citizens and not rely on them...